catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -0,0 +1,445 @@
+    }
+
+    private List<ContentItem> getVersionedContent(Metacard root, List<ContentItem> contentItems,
+            MetacardVersion.Action versionAction, Map<String, Path> tmpContentPaths) {
+        String id = root.getId();
+        MetacardVersion rootVersion = new MetacardVersion(root,
+                versionAction,
+                SecurityUtils.getSubject());
+
+        List<ContentItem> relatedContent = contentItems.stream()
+                .filter(ci -> ci.getId()
+                        .equals(id))
+                .collect(Collectors.toList());
+
+        ContentItem rootItem = relatedContent.stream()
+                .filter(ci -> !isNullOrEmpty(ci.getUri()))
+                .filter(ci -> !ci.getUri()
+                        .contains("#"))
+                .findAny()
+                .orElseThrow(() -> new RuntimeException(
+                        "Could not find root content item for: " + id));
+
+        List<ContentItem> derivedContent = relatedContent.stream()
+                .filter(ci -> !ci.equals(rootItem))
+                .collect(Collectors.toList());
+
+        List<ContentItem> resultItems = new ArrayList<>();
+
+        { // version root content item
+            long size = 0;
+            try {
+                size = rootItem.getSize();
+            } catch (IOException e) {
+                LOGGER.warn("Could not get size of item [{}].", rootItem.getId(), e);
+            }
+            resultItems.add(new ContentItemImpl(rootVersion.getId(),
+                    Files.asByteSource(tmpContentPaths.get(rootItem.getId())
+                            .toFile()),
+                    rootItem.getMimeTypeRawData(),
+                    rootItem.getFilename(),
+                    size,
+                    rootVersion));
+        }
+
+        { // version derived content items
+            for (ContentItem contentItem : derivedContent) {
+                long size = 0;
+                try {
+                    size = contentItem.getSize();
+                } catch (IOException e) {
+                    LOGGER.warn("Could not get size of item [{}].", rootItem.getId(), e);
+                }
+                resultItems.add(new ContentItemImpl(rootVersion.getId(),
+                        contentItem.getQualifier(),
+                        new WrappedByteSource(contentItem),
+                        contentItem.getMimeTypeRawData(),
+                        contentItem.getFilename(),
+                        size,
+                        rootVersion));
+            }
+        }
+
+        return resultItems;
+    }
+
+    private boolean doSkip(@Nullable Operation op) {
+        return doSkip(Optional.ofNullable(op.getProperties())
+                .orElse(Collections.emptyMap()));
+    }
+
+    private boolean doSkip(Map<String, Serializable> properties) {
+        return !historyEnabled || properties.getOrDefault(ALREADY_VERSIONED, false)
```

Comment:
this is nit - but the check for the `historyEnabled` flag should probably be first in the whole expression. You could also condense this to 1 method.